src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -172,13 +172,13 @@ class VariantSampleMatrix[T](val hc: HailContext, val metadata: VariantMetadata,
     val ktRDD = mapPartitionsWithAll { it =>
       it.map { case (v, va, s, sa, g) =>
         keyEC.setAll(localGlobalAnnotation, v, va, s, sa, g)
-        val key = Annotation.fromSeq(keyF().map(_.orNull))
+        val key = Annotation.fromSeq(keyF())
         (key, Annotation(localGlobalAnnotation, v, va, s, sa, g))
       }
     }.aggregateByKey(zVals)(seqOp, combOp)
       .map { case (k, agg) =>
         resultOp(agg)
-        (k, Annotation.fromSeq(aggF().map(_.orNull)))
+        (k, Annotation.fromSeq(aggF()))
```

Comment:
Per your comment earlier, it seems WrappedArray overrides the standard toArray and is fine.  Anything else will copy.